### PR TITLE
Resolve broken links and other minor issues in Chapter 20.

### LINKED ISSUE #219: 20. Resolve broken links and other minor issues

### PR DESCRIPTION
**Pull request by kirtchev-adacore**
Fixes #219.

### CODE REVIEW

File: src/macros.rst [hunk 1]
```diff
@@ -412,10 +425,11 @@ the :t:`token` stream produced from the related :s:`EnumDeclaration`,
 A :t:`derive macro` adds all its declared :t:`[derive helper attribute]s` into
 the :t:`abstract data type scope` of the :t:`abstract data type` the
 :t:`attribute` is attached to.
+IS THIS RELEVANT NOW? WHICH SCOPE?
```
Comment (Veykril) [inline src/macros.rst:428]: Ah, right this is a thing. These are basically visible for attributes on the whole ADT.
```rs
#[derive(MyDerive)]
#[helper_visible_here]
enum Foo {
   #[helper_visible_here]
   Variant {
       #[helper_visible_here]
       field: ()
   }
}

```

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
bors merge

File: src/macros.rst [hunk 2]
```diff
@@ -149,38 +150,46 @@ A :t:`fragment specifier` is a :t:`construct` that indicates the :t:`type` of
 a :t:`metavariable`.
 
 :dp:`fls_8o9mcV2KrKac`
-:t:`[metavariable]s` have :t:`[fragment specifier restriction]s` on what may follow them in a :s:`MacroMatcher`:
+:t:`Fragment specifier` kinds impose the following
+:dt:`[fragment specifier restruction]s` on the :t:`[token]s` that follow them:
```
Comment (riking) [inline src/macros.rst:154]: restruction => restriction

Specify somewhere that these restrictions are on the call site, not the macro definition. The example below contains no macro calls.

Comment (kirtchev-adacore) [reply src/macros.rst:154]: Thanks for catching this, will fix!

Comment (kirtchev-adacore) [reply src/macros.rst:154]: @riking
@Veykril indicated that the current phrasing is actually correct, the restrictions apply to the [macro declaration](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=42e223f70885d65d3a3e8b63b980b730), not the macro invocation.

**Pull request merged**